sdv/constraints/base.py

```diff
@@ -112,30 +121,82 @@ def __init__(self, handling_strategy):
+        if they will not be needing the model to handle conditional sampling.
 
         Args:
             table_data (pandas.DataFrame):
                 Table data.
         """
-        del table_data
+        if self.fit_columns_model and len(self.constraint_columns) > 1:
+            data_to_model = table_data[list(self.constraint_columns)]
+            self._hyper_transformer = HyperTransformer(dtype_transformers={
+                'O': 'one_hot_encoding',
+            })
+            transformed_data = self._hyper_transformer.fit_transform(data_to_model)
+            self._columns_model = GaussianMultivariate()
+            self._columns_model.fit(transformed_data)
+
+        return self._fit(table_data)
 
     def _transform(self, table_data):
         return table_data
 
+    def _sample_constraint_columns(self, table_data):
+        condition_columns = [c for c in self.constraint_columns if c in table_data.columns]
+        grouped_conditions = table_data[condition_columns].groupby(condition_columns)
+        sampled_rows = list()
+        for group, df in grouped_conditions:
+            if not isinstance(group, tuple):
+                group = [group]
+
+            transformed_condition = self._hyper_transformer.transform(df).loc[0].to_dict()
+            sampled_row = self._columns_model.sample(
+                num_rows=df.shape[0],
+                conditions=transformed_condition
+            )
+            sampled_rows.append(sampled_row)
+        sampled_data = pd.concat(sampled_rows)
+        sampled_data = self._hyper_transformer.reverse_transform(sampled_data)
+        return sampled_data
+
     def _validate_constraint_columns(self, table_data):
         """Validate the columns in ``table_data``.
 
-        If any columns in ``constraint_columns`` are not present in ``table_data``,
-        this method will raise a ``MissingConstraintColumnError``.
+        If ``fit_columns_model`` is False and any columns in ``constraint_columns``
+        are not present in ``table_data``, this method will raise a
+        ``MissingConstraintColumnError``. Otherwise it will return the ``table_data``
+        unchanged. If ``fit_columns_model`` is True, then this method will sample
+        any missing ``constraint_columns`` from its model conditioned on the
+        ``constraint_columns`` that ``table_data`` does contain. If ``table_data``
+        doesn't contain any of the ``constraint_columns`` then a
+        ``MissingConstraintColumnError`` will be raised.
 
         Args:
             table_data (pandas.DataFrame):
                 Table data.
         """
-        if any(col not in table_data.columns for col in self.constraint_columns):
-            raise MissingConstraintColumnError()
+        missing_columns = [col for col in self.constraint_columns if col not in table_data.columns]
+        if missing_columns:
+            all_columns_missing = all(c in missing_columns for c in self.constraint_columns)
```

Comment:
I think we could do this as `len(missing_columns) == len(self.constraint_columns)`, so we do not need to iterate again.